src/Microsoft.AspNetCore.Blazor.Build/targets/Blazor.MonoRuntime.targets

```diff
@@ -0,0 +1,305 @@
+      <BlazorItemSpec Include="@(MonoAsmjsFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorRuntimeAsmjsOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>AsmJs</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+      <BlazorItemSpec Include="@(MonoWasmFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorRuntimeWasmOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>WebAssembly</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+      <BlazorItemSpec Include="@(BlazorJsFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorJsOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>BlazorRuntime</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+    </ItemGroup>
+
+    <PropertyGroup Label="Intermediate output paths">
+
+      <!-- /obj/debug/netstandard2.0/blazor -->
+      <BlazorIntermediateOutputPath>$(ProjectDir)$(IntermediateOutputPath)$(BaseBlazorIntermediateOutputPath)</BlazorIntermediateOutputPath>
+
+      <!-- /obj/debug/netstandard2.0/blazor/linker/ -->
+      <BlazorIntermediateLinkerOutputPath>$(BlazorIntermediateOutputPath)linker/</BlazorIntermediateLinkerOutputPath>
+      
+      <!-- /obj/debug/netstandard2.0/blazor/linker.marker.txt -->
+      <BlazorIntermediateLinkerResultFilePath>$(BlazorIntermediateOutputPath)linker.marker.txt</BlazorIntermediateLinkerResultFilePath>
+
+      <!-- /obj/debug/netstandard2.0/blazor/Index.html -->
+      <BlazorIndexHtmlOutputPath>$(BlazorIntermediateOutputPath)$(BlazorIndexHtmlName)</BlazorIndexHtmlOutputPath>
+    
+      <!-- /obj/debug/netstandard2.0/blazor/Descriptor.xml -->
+      <GeneratedBlazorLinkerDescriptor>$(BlazorIntermediateOutputPath)Descriptor.xml</GeneratedBlazorLinkerDescriptor>
+
+      <!-- /obj/debug/netstandard2.0/blazor/Descriptor.xml -->
+      <BlazorResolvedAssembliesOutputPath>$(BlazorIntermediateOutputPath)ResolvedAssemblies.txt</BlazorResolvedAssembliesOutputPath>
+
+    </PropertyGroup>
+    
+    <ItemGroup Label="Different outputs used through the build">
+      <ProducedLinkerDescriptor Include="$(GeneratedBlazorLinkerDescriptor)" />
+      <LinkerMarkerFileOutput Include="$(BlazorIntermediateLinkerResultFilePath)" />
+      <BlazorIndexHtmlFileOutput Include="$(BlazorIndexHtmlOutputPath)" />
+      <BlazorResolvedAssembliesOutput Include="BlazorResolvedAssembliesOutputPath" />
+    
+  </ItemGroup>
+   
+    <MakeDir Directories="$(BlazorIntermediateOutputPath)" />    
+    
+  </Target>
+
+  <Target Name="ResolveBinaryOutput" DependsOnTargets="CollectLinkerOutputs;GetBaseClassLibrary" />
+
+  <Target Name="CollectLinkerDescriptors"
+          Inputs="@(IntermediateAssembly);@(BlazorLinkerDescriptor);@(ProducedLinkerDescriptor)"
+          Outputs="@(ProducedLinkerDescriptor)"
+          Condition="$(BlazorShouldRunMonoLinker) == 'true' and '@(BlazorLinkerDescriptor)' == ''">
```

Comment:
What's the purpose of `'@(BlazorLinkerDescriptor)' == ''` in the condition? That's seems a little strange

Reply:
That you haven't provided one explicitly

Reply:
Essentially if you give me a descriptor file I'll use it, if not I'll generate one to inform the linker to preserve all the code in the main app assembly.